### PR TITLE
Upgrade Reactor and Reactor Netty to Latest Patch

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR updates Reactor and Reactor Netty to their latest patch versions of `3.3.x` and `0.9.x` to pick up key bug fixes. 

Additionally, `xml-maven-plugin` is upgraded to resolve build warning when using parallel build. It's configuration is moved to a profile that validates the target file exists before running to reduce the number of times the plugin runs if Spotbugs is skipped.

### CODE REVIEW

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -56,8 +56,8 @@ io.netty:netty-tcnative-boringssl-static;2.0.34.Final
 io.netty:netty-transport-native-epoll;4.1.53.Final
 io.netty:netty-transport-native-unix-common;4.1.53.Final
 io.netty:netty-transport-native-kqueue;4.1.53.Final
-io.projectreactor.netty:reactor-netty;0.9.13.RELEASE
-io.projectreactor:reactor-core;3.3.11.RELEASE
+io.projectreactor.netty:reactor-netty;0.9.15.RELEASE
```
Comment (kushagraThapar) [inline eng/versioning/external_dependencies.txt:59]: No updates on netty versions ?
This release uses 54.Final versions of netty. 
https://mvnrepository.com/artifact/io.projectreactor.netty/reactor-netty/0.9.15.RELEASE

**Review by kushagraThapar — CHANGES_REQUESTED** [review]

File: eng/versioning/external_dependencies.txt (hunk 1 above)
Comment (alzimmermsft) [reply eng/versioning/external_dependencies.txt:59]: Good catch

Comment (kushagraThapar) [reply eng/versioning/external_dependencies.txt:59]: LGTM

**Review by kushagraThapar — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

**Pull request merged**